src/carbonado.rs

```diff
@@ -114,12 +114,20 @@ pub async fn retrieve(sk: &str, name: &str) -> Result<Vec<u8>> {
 
 #[cfg(feature = "server")]
 pub async fn retrieve(sk: &str, name: &str) -> Result<Vec<u8>> {
+    use crate::constants::NETWORK;
+
     let sk = hex::decode(sk)?;
     let secret_key = SecretKey::from_slice(&sk)?;
     let public_key = PublicKey::from_secret_key_global(&secret_key);
     let pk = public_key.to_hex();
 
-    let filepath = handle_file(&pk, name, 0).await?;
+    let network = NETWORK.read().await.to_string();
+    let mut final_name = name.to_string();
+    if !name.contains(&network) {
+        final_name = format!("{network}-{name}");
+    }
+
+    let filepath = handle_file(&pk, &final_name, 0).await?;
```

Comment:
I'm not sure that is either, since these aren't actually called by the server as far as I know, and even if they were, the name is provided by the client, which is where we accounted for this. I actually don't really like these conditionally-compiled methods, from what I understand, they're distracting and unused.